ballerina/tests/from_json_test.bal

```diff
@@ -67,6 +67,55 @@ isolated function testSimpleJsonToRecord() returns Error? {
     test:assertEquals(recC.get("b"), 1);
 }
 
+type ReadOnlyUserRecord readonly & record {|
+    int id;
+|};
+
+@test:Config
+isolated function testSimpleJsonToRecord2() returns Error? {
+    json user = {id: 4012};
+    ReadOnlyUserRecord r = check parseAsType(user);
+    test:assertEquals(r, {id: 4012});
+}
+
+public type ReadonlyUserId readonly & int;
+
+public type ReadonlyUserName readonly & record {
+    string firstname;
+    string lastname;
+};
+
+type ReadOnlyUserRecord2 readonly & record {|
```

Comment:
Will you be able to verify this type as well when the record field has `readonly`
```ballerina
type DDPerson record {|
    readonly int id;
|};
```

Reply:
Shall we create another issue for this. Because this is regarding field readonly, but this PR focus on type readonly.
I will update it in a seperate PR

Reply:
https://github.com/ballerina-platform/ballerina-library/issues/7259

Created a issue for this, will track this issue in there